substrate/primitives/api/test/tests/runtime_calls.rs

```diff
@@ -105,9 +105,12 @@ fn record_proof_works() {
 	.into_unchecked_extrinsic();
 
 	// Build the block and record proof
-	let mut builder = client
-		.new_block_at(client.chain_info().best_hash, Default::default(), true)
-		.expect("Creates block builder");
+	let mut builder = BlockBuilderBuilder::new(&client)
+		.on_parent_block(&client, client.chain_info().best_hash)
```

Comment:
Why do we pass `&client` to `on_parent_block` while it's already passed to `new` and `on_parent_block` takes only one argument (hash)?

Reply:
An oversight :D I had refactored it in the lifetime of the pr :D